src/TesApi.Web/Runner/TaskToNodeTaskConverter.cs

```diff
@@ -0,0 +1,417 @@
+        /// If the path is a local path, the storage account name is extracted from the path.
+        /// If the storage account is a configured external storage account, the SAS token from the configuration is added to the URL.
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="defaultStorageAccountName"></param>
+        /// <returns></returns>
+        private TesInput PrepareLocalFileInput(TesInput input, string defaultStorageAccountName)
+        {
+            //When Cromwell runs in local mode with a Blob FUSE drive, the URL property may contain an absolute path.
+            //The path must be converted to a URL. For Terra this scenario doesn't apply. 
+            if (StorageUrlUtils.IsLocalAbsolutePath(input.Url))
+            {
+                var convertedUrl = StorageUrlUtils.ConvertLocalPathOrCromwellLocalPathToUrl(input.Url, defaultStorageAccountName);
+
+                var blobUrl = AppendSasTokenIfExternalAccount(convertedUrl);
+
+                return new TesInput()
+                {
+                    Name = input.Name,
+                    Description = input.Description,
+                    Path = input.Path,
+                    Url = blobUrl,
+                    Type = input.Type
+                };
+            }
+
+            return default;
+        }
+
+        private async Task<TesInput> UploadContentAndCreateTesInputAsync(TesTask tesTask, string inputPath,
+            string content,
+            CancellationToken cancellationToken)
+        {
+            var inputFileUrl =
+                await storageAccessProvider.GetInternalTesTaskBlobUrlAsync(tesTask, Guid.NewGuid().ToString(),
+                    cancellationToken);
+
+            //return the URL without the SAS token, the runner will add it using the transformation strategy
+            await storageAccessProvider.UploadBlobAsync(new Uri(inputFileUrl), content, cancellationToken);
+
+            var inputUrl = StorageUrlUtils.RemoveQueryStringFromUrl(inputFileUrl);
+
+            logger.LogInformation($"Successfully uploaded content input as a new blob at: {inputUrl}");
+
+            return new TesInput
+            {
+                Path = inputPath,
+                Url = inputUrl,
+                Type = TesFileType.FILEEnum,
+            };
+        }
+
+        private async Task<TesInput> PrepareContentInputAsync(TesTask tesTask, TesInput input,
+            CancellationToken cancellationToken)
+        {
+
+            if (String.IsNullOrWhiteSpace(input?.Content))
+            {
+                return default;
+            }
+
+            logger.LogInformation($"The input is content. Uploading its content to the internal storage location. Input path:{input.Path}");
+
+            return await UploadContentAndCreateTesInputAsync(tesTask, input.Path, input.Content, cancellationToken);
+        }
+
+        private static string EscapeBashArgument(string arg)
```

Comment:
Are you dumping the commands from the TesTask into a script in a single string or are you passing them to the Docker API? If the latter, we should NOT be shell escaping them. The Docker API does not turn them into a single string command-line, it passes them to the process's main() function directly.